src/Data/Aeson/TypeScript/Types.hs

```diff
@@ -98,13 +98,16 @@ data FormattingOptions = FormattingOptions
   -- ^ Function applied to generated interface names
   , typeNameModifier :: String -> String
   -- ^ Function applied to generated type names
+  , exportTypes :: Bool
```

Comment:
In general this PR sounds good to me. I'm wondering if we should reduce boolean blindness / provide for future extensibility by making this `data` type though. Are there any other kinds of exporting someone might want to do? I know that there's some fancy stuff [here](https://www.typescriptlang.org/docs/handbook/modules.html) like `export default`, `export =`, `export as namespace`... none of them seem very likely to be desired but it would be good to be safe.

Reply:
Wow, that was fast :) Thanks a lot for the quick review.

- `export default` => I personally don't have a use case for this, but it might be useful in some cases. Introducing a "default type" might require a bit more work API-wise. I guess we should expose a way for the user to tell which `Name` is supposed to be the default one. I'm not really willing to implement this for now.
- `export as namespace` => Could be used to nest the "module name". ie `module.namespace.type` instead of `module.type`. It might be useful in some niche cases, I personally don't have any use for this, I'm not sure whether or not we should implement that. I'll let your be the judge here.
- `export = TYPE` => It's basically a synonym of `export TYPE = (...)`

> I'm wondering if we should reduce boolean blindness / provide for future extensibility by making this data type though.

I'm sure I understand what you mean here. Something the lines of:

```haskell
newtype ExportTypes = ExportTypes Bool
newtype ExportInterfaces = ExportInterfaces Bool
data ExportOptions = ExportOptions ExportTypes ExportInterfaces
```

?